### PR TITLE
chore(deps-dev): bump jest cli and some other deps to remove warnings

### PR DESCRIPTION
**Pull request by edleeks87**
### Proposed behaviour

There are a very large number a minor and medium warnings, this pr fixes all the non-breaking ones, it reduces them from over 70k to just 2

### Current behaviour

There over 70k warnings raised

### Checklist

- [x] Release notes (Conventional Commits) 
<del>- [ ] Unit tests<del>
<del>- [ ] Cypress automation tests<del>
<del>- [ ] Storybook added or updated<del>
<del>- [ ] Theme support<del>
<del>- [ ] Typescript `d.ts` file added or updated<del>

### Testing instructions

Main focus is to ensure the upgrade of `cypress` continues to work and then quick sanity check that the components still render in storybook and codesandbox

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,38 +85,39 @@
     "@types/react": "16.9.31",
     "@types/react-dom": "16.0.11",
     "babel-eslint": "10.0.1",
-    "babel-jest": "25.2.6",
+    "babel-jest": "24.9.0",
```
Comment (mkrds) [inline package.json:88]: Are you sure version of `babel-jest` is supposed to be downgraded?

Comment (edleeks87) [reply package.json:88]: Good spot, I'll revert that, I ran `audit --fix` after going through the obvious ones not sure why that got downgraded

Comment (edleeks87) [reply package.json]: https://github.com/Sage/carbon/commit/9e9c06ca20905858e4cd11e3e5d0b0db3280171d

**Pull request merged**